app/controllers/posts_controller.rb

```diff
@@ -17,13 +17,14 @@ def new
 
   def create
     @users = User.find_by(id: params[:user_id])
-    @post = Post.new(user_id: @users.id, title: params[:post][:title], text: params[:post][:text])
+    @post = Post.new(user_id: @users.id, title: params[:post][:title], text: params[:post][:text], comments_counter: 0,
+                     likes_counter: 0)
     if @post.save
       flash[:success] = 'New post created successfully!'
-      redirect_to users_path
+      redirect_to "/users/#{@users.id}/posts"
     else
-      flash.now[:error] = 'Error: Data was not saved!'
-      render :new
+      flash[:error] = @post.errors.full_messages[0]
+      redirect_to "/users/#{@users.id}/posts/new"
     end
   end
```

Comment:
- [ ] I suggest that you solve the n+1 problem when fetching all posts and their comments  you can simply add in your index method `@posts = @user.posts.includes(:comments)` currently to load a user posts page and related comments is a problem
![query](https://user-images.githubusercontent.com/44745855/155833917-625051d6-70bc-4618-97bd-fcc91436281f.png)

Reply:
@DeliceLydia  Thank you very much for the feedback, I'll work on it!